src/audio/kpb.c

```diff
@@ -50,6 +50,10 @@ static const struct comp_driver comp_kpb;
 DECLARE_SOF_UUID("kpb", kpb_uuid, 0xd8218443, 0x5ff3, 0x4a4c,
 		 0xb3, 0x88, 0x6c, 0xfe, 0x07, 0xb9, 0x56, 0x2e);
 
+/* e50057a5-8b27-4db4-bd79-9a639cee5f50 */
+DECLARE_SOF_UUID("kpb-task", kpb_task_uuid, 0xe50057a5, 0x8b27, 0x4db4,
+		 0xbd, 0x79, 0x9a, 0x63, 0x9c, 0xee, 0x5f, 0x50);
+
 /* KPB private data, runtime data */
```

Comment:
I don't know much about UUID. How did you generate this number ?

Reply:
I am using uuidgen tool, part of util-linux available from https://www.kernel.org/pub/linux/utils/util-linux/

Reply:
There are various algorithms to generate UUIDs, what algorithm in particular was used is unimportant as long as it is used correctly. Then the UUID itself is translated to the SOF representation in a way that is reasonably understandable.

Reply:
@mmaka1 @paulstelian97 thanks!

Reply:
Can we include this in the commit message, i.e. how we generate the UUIDs. This is also worth adding a paragraph in sof-docs section for adding new component etc.

Reply:
@lgirdwood Yes, I am working on the component documentation updates and will be adding a short guide to UUIDs in fw as well.

Reply:
@lgirdwood @paulstelian97 @dbaluta Uuid details and generation examples are already in the code (uuid.h + pending sof-docs pr). Do you have any other comments to this one? There are no approvals yet.

Reply:
@mmaka1 My only issue with this is whether we need the UUIDs at runtime (currently the actual values are stripped and we only can use handles to the UUID). Will we ever need the UUIDs themselves at runtime? (both firmware and kernel side to be considered)

Reply:
- @paulstelian97 I hope we will use them soon to create component instances. But I believe that there is no need to keep a full value in the memory.

I planned to create log filtering api based on those 0x1ffff... ids from from the .ldc file since the logger is able to resolve it both ways. So there would be simple uint32_t compare inside the filtering logic in the fw.

Once the topology is switched to uuids for component instances, the driver could resolve the uiuds to 0x1fff.... values too and the fw manifest seems to be a quite good place to publish this information. Btw, having so wide separation between 0x1fff... and current sof_ipc_comp::type, there would be just a tiny adjustment in fw to overload that ipc msg field to make comp_new() working in dual mode and be backward compatible ;)

Reply:
@mmaka1 Yeah, still we will need to adjust which section UUIDs will stay in, when they become important at runtime the kernel must be aware of them, because the "handles" may differ every time we recompile.